### PR TITLE
fix: on admin course users previous user show on the modal

### PR DESCRIPTION
**Pull request by Zoro444**
**🟢 Add `deploy` label if you want to deploy this Pull Request to staging environment**

#### 🧑‍⚖️ Pull Request Naming Convention

- Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- Do not put issue id in title
- Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
- Consider to add `area:*` label(s)

* [x] I followed naming convention rules

---

#### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link
https://github.com/rolling-scopes/rsschool-app/issues/2035

_Describe the source of requirement, like related issue link._

#### 💡 Background and solution

_Describe the big picture of your changes here_

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Database migration is added or not needed
- [x] Documentation is updated/provided or not needed
- [x] Changes are tested locally

### CODE REVIEW

File: client/src/components/Forms/ModalForm.tsx [hunk 1]
```diff
@@ -23,7 +23,9 @@ export function ModalForm(props: Props) {
   if (props.data == null) {
     return null;
   }
-  const initialValues = props.getInitialValues ? props.getInitialValues?.(props.data) : props.data;
+  const formValues = props.getInitialValues ? props.getInitialValues?.(props.data) : props.data;
+  form.setFieldsValue(formValues);
```
[the remark this answers is not in the document]
Comment (Zoro444) [reply client/src/components/Forms/ModalForm.tsx:27]: I read this in ant design documentation

isn't that the problem?

"
You cannot set value for each form control via value or defaultValue prop, you should set default value with initialValues of Form. Note that initialValues cannot be updated by setState dynamically, you should use setFieldsValue in that situation.

You shouldn't call setState manually, please use form.setFieldsValue to change value programmatically.
"

[the remark this answers is not in the document]
Comment (aaliakseyenka) [reply client/src/components/Forms/ModalForm.tsx:27]: sure, what antd says is ok. The problem is that you make form updates each render cycle, which is basically side effect, and that shouldn't be done in react

File: client/src/components/Forms/ModalForm.tsx [hunk 2]
```diff
@@ -38,6 +40,7 @@ export function ModalForm(props: Props) {
           return;
         }
         props.submit(values);
+        form.resetFields();
```
Comment (apalchys) [inline client/src/components/Forms/ModalForm.tsx:43]:
```suggestion
        form.resetFields2();
```

**Comment by apalchys** [comment]
@Zoro444 The root cause of the issue lays in our Modal with Form is rendered. Initially it was a mistake to design base FormModal component in that way. 

Quick and dump solution is just do not render Modal at all if we have no data. I just created a MR that should address it: https://github.com/rolling-scopes/rsschool-app/pull/2443

Please check and review.

cc @valerydluski

**Pull request closed without merging**